omegaconf/omegaconf.py

```diff
@@ -261,14 +263,16 @@ def load(file_: Union[str, pathlib.Path, IO[Any]]) -> Union[DictConfig, ListConf
 
     @staticmethod
     def save(
-        config: Container, f: Union[str, pathlib.Path, IO[Any]], resolve: bool = False
+        config: Any, f: Union[str, pathlib.Path, IO[Any]], resolve: bool = False
     ) -> None:
         """
         Save as configuration object to a file
         :param config: omegaconf.Config object (DictConfig or ListConfig).
         :param f: filename or file object
         :param resolve: True to save a resolved config (defaults to False)
         """
+        if is_dataclass(config) or is_attr_class(config):
+            config = OmegaConf.create(config)
```

Comment:
I was actually thinking about just accepting DictConfig annotated as something else here, but this works too.